pyomo/core/base/componentuid.py

```diff
@@ -76,17 +77,21 @@ class ComponentUID(object):
     def __init__(self, component, cuid_buffer=None, context=None):
         # A CUID can be initialized from either a reference component or
         # the string representation.
+        def _context_err(_type):
+            raise ValueError(
+                f"Context is not allowed when initializing a ComponentUID from {_type}."
+            )
         if isinstance(component, str):
             if context is not None:
-                raise ValueError(
-                    "Context is not allowed when initializing a "
-                    "ComponentUID object from a string type"
-                )
+                _context_err(str)
             try:
                 self._cids = tuple(self._parse_cuid_v2(component))
             except (OSError, IOError):
                 self._cids = tuple(self._parse_cuid_v1(component))
-
+        elif isinstance(component, ComponentUID):
```

Comment:
In `find_component`, we used `type is ComponentUID` instead of `isinstance`, presumably for performance. Should we do the same here?